{
  for (  ExecutionSiteRunner r : m_runners) {
    r.m_shouldStartRunning.countDown();
  }
  if (m_restoreAgent != null) {
    m_restoreAgent.restore();
  }
 else {
    onRestoreCompletion(Long.MIN_VALUE);
  }
  if (m_rejoinCoordinator != null) {
    m_rejoinCoordinator.startRejoin();
  }
  Thread.currentThread().setName("ExecutionSiteAndVoltDB");
  m_isRunning=true;
  try {
    m_currentThreadSite.run();
  }
 catch (  Throwable thrown) {
    String errmsg=" encountered an unexpected error and will die, taking this VoltDB node down.";
    hostLog.error(errmsg);
    hostLog.fatal("Stack trace of thrown exception: " + thrown.toString());
    for (    StackTraceElement ste : thrown.getStackTrace()) {
      hostLog.fatal(ste.toString());
    }
    VoltDB.crashLocalVoltDB(errmsg,true,thrown);
  }
}
